### PR TITLE
PMK-2258: Added "Finish And Review" button for AWS and Azure cluster creation.

### PR DESCRIPTION
**Pull request by rahulkurumkar**

### CODE REVIEW

File: src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js [hunk 1]
```diff
@@ -44,6 +44,7 @@ const initialContext = {
   runtimeConfigOption: 'default',
   isPrivate: false,
   internalElb: false,
+  showFinishAndReview: true,
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js]: We definitely don't want to mix this into the `wizardContext` as an ordinary value.  We should use a predicate function we can pass as a prop into the Wizard component.

Something like:

```typescript
const canReview = (wizardContext): boolean => hasAllRequiredFields(ctx)
<Wizard showFinishAndReview={canReview} />
```

Comment (rahulkurumkar) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js]: Passing `canReview` props to `<Wizard />` as the validation part is already taken care.

File: src/app/core/components/wizard/Wizard.js [hunk 2]
```diff
@@ -90,7 +110,7 @@ class Wizard extends PureComponent {
 
   render () {
     const { wizardContext, setWizardContext, steps, activeStep } = this.state
-    const { showSteps, children, submitLabel, onCancel } = this.props
+    const { showSteps, children, submitLabel, finishAndReviewLabel, onCancel, canReview } = this.props
```
Comment (xantorres) [inline src/app/core/components/wizard/Wizard.js]: `canReview` is a bit strange (seems something about permissions), perhaps `showFinishButton` is more self explanatory?

File: src/app/core/components/wizard/Wizard.js [hunk 3]
```diff
@@ -116,12 +137,14 @@ Wizard.propTypes = {
   onCancel: PropTypes.func,
   context: PropTypes.object,
   submitLabel: PropTypes.string,
+  finishAndReviewLabel: PropTypes.string,
```
Comment (xantorres) [inline src/app/core/components/wizard/Wizard.js]: missing the new `canReview` (or whichever name you end up calling it) prop here

File: src/app/core/components/wizard/Wizard.js [hunk 4]
```diff
@@ -71,6 +72,25 @@ class Wizard extends PureComponent {
...
+    const { activeStep } = this.state
+
+    if (this.nextCb[activeStep] && this.nextCb[activeStep]() === false) {
+      return
+    }
+
+    this.setState(
+      state => ({ activeStep: state.steps.length - 1 }),
+      () => {
+        this.activateStep()
```
Comment (xantorres) [inline src/app/core/components/wizard/Wizard.js]: I'm not sure why we are calling `this.activateStep` and therefore doing another `setState` after setting the active step. It looks like something that could be achieved simply by updating the state once (and avoiding one extra useless re-render).
I know we do the same when navigating between contiguous  steps, but still, something that could be reviewed and improved.

File: src/app/core/components/wizard/Wizard.js (hunk 2 above)
Comment (rahulkurumkar) [reply src/app/core/components/wizard/Wizard.js]: Changed it to `showFindAndReviewButton`

File: src/app/core/components/wizard/Wizard.js (hunk 4 above)
Comment (rahulkurumkar) [reply src/app/core/components/wizard/Wizard.js]: Refactored the code to update activeStepId and activeStep in single go

File: src/app/core/components/wizard/Wizard.js (hunk 3 above)
Comment (rahulkurumkar) [reply src/app/core/components/wizard/Wizard.js]: Added

**Pull request merged**